### ISSUE TITLE
DataLoader Problem

### ISSUE THREAD

**Issue by suchithtuple** [pytorch/pytorch#20701]
I have training , validation and test dataset. I have created dataloaders for three of them with training shuffle = True and for valdiation and test shuffle = False . So in one of the code epoch loop contains iterator over train and validation and in another one , epoch loop contains iterator over train , validation and test. 

It's obvious that train example id's must be same in both the cases but they are different. 

The code is given below :

-  train + validation : https://www.kaggle.com/suchith0312/pytorch-dataloader-testing?scriptVersionId=14425205

- train + validation + test : https://www.kaggle.com/suchith0312/pytorch-dataloader-testing?scriptVersionId=14425182

**Comment by soumith** [comment]
please use https://discuss.pytorch.org for questions

**Comment by soumith** [comment]
if this is a bug report, format it according to our bug report template, and give a small snippet of code that reproduces the bug

**Comment by suchithtuple** [comment]
I formatted according to the template and raise the issue here : https://github.com/pytorch/pytorch/issues/20717

Is there anything other than this do let me know.  Apologies for the inconvenience caused.
